### ISSUE TITLE
[PHP8.3] "final public const string JOURNEY" leads to "Class constants must be uppercase"

### ISSUE THREAD

**Issue by ralusnom** [squizlabs/PHP_CodeSniffer#3936]
## Description

In IntelliJ IDEA phpcs, incorrectly flags valid PHP code as an warning. Specifically, it misinterprets the type of class constants. For example, declaring a constant like `final public const string JOURNEY;` results in an error message from PHP CS Fixer stating 

"phpcs: Class constants must be uppercase; expected STRING but found string".

It seems that PHP CS Fixer is incorrectly checking the type declaration (`string`) instead of the constant name (`JOURNEY`).

## Runtime version

PHP CS Fixer v3.40.2, when used with PHP 8.3 

## Code snippet that reproduces the problem

```php
<?php
// Erroneous code snippet
final public const string JOURNEY;
```

**Comment by jrfnl** [comment]
Closing as reported to the wrong project. [CS-Fixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer) !== PHP_CodeSniffer.

**Comment by kubawerlos** [comment]
@jrfnl actually reported in the correct project, only reported wrongly, see the error message.

**Comment by jrfnl** [comment]
FYI: a fix for this issue is included in today's [PHP_CodeSniffer 3.9.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.9.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
